src/main/java/org/folio/rest/impl/TenantRefAPI.java

```diff
@@ -25,32 +28,27 @@ private String filter(String content) {
     return jOutput.encodePrettily();
   }
 
+  @Validate
   @Override
-  public void postTenant(TenantAttributes ta, Map<String, String> headers,
-    Handler<AsyncResult<Response>> hndlr, Context cntxt) {
-    log.info("postTenant");
-    Vertx vertx = cntxt.owner();
-    super.postTenant(ta, headers, res -> {
-      if (res.failed()) {
-        hndlr.handle(res);
-        return;
-      }
-      TenantLoading tl = new TenantLoading();
-      tl.withKey("loadSample").withLead("sample-data")
-        .withPostOnly()
-        .withFilter(this::filter)
-        .withAcceptStatus(422)
-        .add("users", "authn/credentials")
-        .perform(ta, headers, vertx, res1 -> {
-          if (res1.failed()) {
-            hndlr.handle(io.vertx.core.Future.succeededFuture(PostTenantResponse
-              .respond500WithTextPlain(res1.cause().getLocalizedMessage())));
-            return;
-          }
-          hndlr.handle(io.vertx.core.Future.succeededFuture(PostTenantResponse
-            .respond201WithApplicationJson("")));
+  public void postTenant(TenantAttributes tenantAttributes, Map<String, String> headers,
+      Handler<AsyncResult<Response>> handler, Context context) {
+    super.postTenantSync(tenantAttributes, headers, handler, context);
+  }
+  
```

Comment:
You should not override postTenant and call postTenantSync. If modules does that, then it will never be async. So please don't. That means you'll have to update your tests, since they are HTTP based. Or you call postTenantSync straight from your tests (without HTTP).

Reply:
@adamdickmeiss Can we add a test wrapper for the HTTP get, something like
int resp;
do {
   resp = doGet;
   wait 5;
} while (resp != 204;

So people can easilly switch over their unit test to the async version

Reply:
Updated all tests to call `postTenantSync` directly.